src/components/embedded/EmbeddedMessage.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useEffect, useState } from 'react';
+import api from '../../api';
+import { EventMessage, MessageViewType } from '../../models/EventMessage';
+import { MinimalMessageCard } from '../message/message-card/MessageCard';
+import SplashScreen from '../SplashScreen';
+
+function EmbeddedMessage({ messageId }: { messageId: string }) {
+	const [message, setMessage] = useState<EventMessage | null>();
+	const [viewType, setViewType] = useState(MessageViewType.JSON);
+
+	useEffect(() => {
+		getMessage();
+	}, []);
+
+	async function getMessage() {
+		const res = await api.messages.getMessage(messageId);
+		setMessage(res);
+		setViewType(message?.viewType ? message.viewType : MessageViewType.JSON);
```

Comment:
actually viewType doesn't exist in message i think. we should delete that property from interface